hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCallHandlerTest.java

```diff
@@ -80,11 +85,19 @@ class ContractCallHandlerTest extends ContractHandlerTestBase {
     @Mock
     private GasCalculator gasCalculator;
 
+    @Mock(strictness = Strictness.LENIENT)
+    private ContractServiceComponent contractServiceComponent;
+
+    private final Metrics metrics = new NoOpMetrics();
+    private final ContractMetrics contractMetrics = new ContractMetrics();
+
     private ContractCallHandler subject;
 
     @BeforeEach
     void setUp() {
-        subject = new ContractCallHandler(() -> factory, gasCalculator);
+        contractMetrics.init(metrics);
+        given(contractServiceComponent.contractMetrics()).willReturn(contractMetrics);
+        subject = new ContractCallHandler(() -> factory, gasCalculator, contractServiceComponent);
```

Comment:
I'm surprised not to see a test verifying that metrics are invoked in an error case, for any of the handlers changed. Is there a separate plan for catching future regressions?

Reply:
I don't know how to test it with HapiSpec either, which I'd really like to do.

w.r.t. unit test you're right - I meant to do that by subclassing the metrics instance and injecting that.  But it'll work better with your suggestion above of a MetricsContext.